File: shoring/compat/src/main/java/com/panopset/compat/FileTransformer.java
package com.panopset.compat;

import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.File;
import java.io.FileReader;
import java.io.FileWriter;
import java.io.IOException;

import static com.panopset.compat.Stringop.getEol;

public abstract class FileTransformer {

  protected abstract String filter(String inputLine);

  public FileTransformer(File input, File output) {
    inp = input;
    outp = output;
  }

  public void exec() {
    try (FileReader fr = new FileReader(inp);
      BufferedReader br = new BufferedReader(fr);
      FileWriter fw = new FileWriter(outp);
      BufferedWriter bw = new BufferedWriter(fw)
    ) {
      String line = br.readLine();
      while (line != null) {
        String fl = filter(line);
        if (fl != null) {
          bw.write(fl);
          bw.write(getEol());
        }
        line = br.readLine();
      }
    } catch (IOException e) {
      Logop.errorEx(e);
    }
  }

  private final File inp;

  private final File outp;
}

File: shoring/compat/src/main/java/com/panopset/compat/FileTransformer.kt
package com.panopset.compat

import com.panopset.compat.Logop.errorEx
import com.panopset.compat.Stringop.getEol
import java.io.*

abstract class FileTransformer(private val panop: Panop, private val inp: File, private val outp: File) {
    protected abstract fun filter(inputLine: String): String

    fun exec() {
        try {
            FileReader(inp).use { fr ->
                BufferedReader(fr).use { br ->
                    FileWriter(outp).use { fw ->
                        BufferedWriter(fw).use { bw ->
                            var line = br.readLine()
                            while (line != null) {
                                val fl = filter(line)
                                bw.write(fl)
                                bw.write(getEol())
                                line = br.readLine()
                            }
                        }
                    }
                }
            }
        } catch (e: IOException) {
            errorEx(panop, e)
        }
    }
}
